### ISSUE TITLE
token versioning

### ISSUE THREAD

**Issue by iraj720** [golang-jwt/jwt#421]
i have added versioning to the library. i think it makes it easier since we are facing with this challenge.

also it doesn't effect other parts of the projects.

here it is : https://github.com/golang-jwt/jwt/pull/420

**Comment by iraj720** [comment]
yes i can but i need to rewrite it for all services and i think it might be useful for others also. do you agree with it?

**Comment by mfridman** [comment]
We have a finite capacity to integrate additional logic into the core library. If we were to fulfill every feature request we receive, it could lead to considerable bloat within the package, increasing the API surface and harder to maintain. 

We're trying to strike a balance to ensure optimal user experience. Let's see if @oxisto has an opinion here.
